controllers/comment.controller.js

```diff
@@ -0,0 +1,188 @@
+const Comment = require('../models/comment.model');
+const Post = require('../models/post.model');
+const User = require('../models/user.model');
+
+const addCommentToPostAndUser = async (postID, userID, commentID) => {
+  try {
+    const user = await User.findById(userID);
+    user.comments = [...user.comments, commentID];
+    await user.save();
+    const post = await Post.findById(postID);
+    post.comments = [...post.comments, commentID];
+    await post.save();
+  } catch (error) {
+    return error.message;
+  }
+};
+
+const addComment = async (req, res) => {
+  const { content, postID } = req.body;
+  const { id } = req.user;
```

Comment:
These changes should be applied if comment model suggestion is also applied (see `models/comment.model.js`)

```suggestion
  const { id:userID } = req.user;
```